### PR TITLE
Clean up some http.sys code

### PR DESCRIPTION
**Pull request by davidfowl**
- Remove SuppressContext and ToIAsyncResult helpers.
Use TaskToApm instead.
- Dispatch off the current thread to start the accept loop
- Use logging pattern in ResponseBody
- Replace Contract.Assert with Debug.Assert
- Use new ValidateBufferArguments to verify WriteAsync overloads

### CODE REVIEW

File: src/Servers/HttpSys/src/RequestProcessing/ResponseBody.cs [hunk 1]
```diff
@@ -733,5 +734,72 @@ private void CheckDisposed()
                 throw new ObjectDisposedException(GetType().FullName);
             }
         }
+
+        private class Log
```
Comment (JamesNK) [inline src/Servers/HttpSys/src/RequestProcessing/ResponseBody.cs:738]:
```suggestion
        private static class Log
```

File: src/Servers/HttpSys/src/RequestProcessing/ResponseBody.cs [hunk 2]
```diff
@@ -733,5 +734,72 @@ private void CheckDisposed()
...
+            }
+
+            public static void WriteFlushCancelled(ILogger logger, uint statusCode)
+            {
+                _writeFlushCancelled(logger, statusCode, null);
+            }
+
+            public static void FileSendAsyncError(ILogger logger, Exception e)
+            {
+                _fileSendAsyncError(logger, e);
```
Comment (JamesNK) [inline src/Servers/HttpSys/src/RequestProcessing/ResponseBody.cs:791]: nit
```suggestion
            public static void FileSendAsyncError(ILogger logger, Exception exception)
            {
                _fileSendAsyncError(logger, exception);
```

File: src/Servers/HttpSys/src/MessagePump.cs [hunk 3]
```diff
@@ -134,7 +134,8 @@ public Task StartAsync<TContext>(IHttpApplication<TContext> application, Cancell
                 _serverAddresses.Addresses.Add(prefix.FullPrefix);
             }
 
-            ActivateRequestProcessingLimits();
+            // Dispatch to get off the SynchronizationContext and use UnsafeQueueUserWorkItem to avoid capturing the ExecutionContext
+            ThreadPool.UnsafeQueueUserWorkItem(state => state.ActivateRequestProcessingLimits(), this, preferLocal: false);
```
Comment (jkotalik) [inline src/Servers/HttpSys/src/MessagePump.cs:138]: Possible to drop the async void here? https://github.com/dotnet/aspnetcore/blob/b795ac3546eb3e2f47a01a64feb3020794ca33bb/src/Servers/HttpSys/src/MessagePump.cs#L177

Comment (davidfowl) [reply src/Servers/HttpSys/src/MessagePump.cs:138]: Yes async void is 😄

Comment (jkotalik) [reply src/Servers/HttpSys/src/MessagePump.cs:138]: thx

**Review by jkotalik — APPROVED** [review]

File: src/Servers/HttpSys/src/RequestProcessing/Request.cs [hunk 4]
```diff
@@ -370,7 +370,7 @@ public async Task<X509Certificate2> GetClientCertificateAsync(CancellationToken
             var certLoader = new ClientCertLoader(RequestContext, cancellationToken);
             try
             {
-                await certLoader.LoadClientCertificateAsync().SupressContext();
+                await certLoader.LoadClientCertificateAsync();
```
Comment (Tratcher) [inline src/Servers/HttpSys/src/RequestProcessing/Request.cs:373]: GetClientCertificateAsync is called from app code, you might still want the suppression here.

**Review by Tratcher — APPROVED** [review]

File: src/Servers/HttpSys/src/RequestProcessing/Request.cs (hunk 4 above)
Comment (davidfowl) [reply src/Servers/HttpSys/src/RequestProcessing/Request.cs:373]: We don't do it *anywhere* in ASP.NET Core, so this isn't special.

**Pull request merged**